src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/ObjectWriter/CodeView/CodeViewTypesBuilder.cs

```diff
@@ -0,0 +1,537 @@
+                TypeFlags.Int64 => T_INT8,
+                TypeFlags.UInt64 => T_UINT8,
+                TypeFlags.IntPtr => _targetPointerSize == 8 ? T_INT8 : T_INT4,
+                TypeFlags.UIntPtr => _targetPointerSize == 8 ? T_UINT8 : T_UINT4,
+                TypeFlags.Single => T_REAL32,
+                TypeFlags.Double => T_REAL64,
+                _ => T_NOTYPE,
+            });
+        }
+
+        public uint GetPointerTypeIndex(PointerTypeDescriptor pointerDescriptor)
+        {
+            using (LeafRecordWriter record = StartLeafRecord(LF_POINTER))
+            {
+                record.Write(pointerDescriptor.ElementType);
+                record.Write((uint)(
+                    (pointerDescriptor.Is64Bit == 1 ? CV_PTR_64 : CV_PTR_NEAR32) |
+                    (pointerDescriptor.IsReference == 1 ? CV_PTR_MODE_LVREF : CV_PTR_MODE_PTR) |
+                    (pointerDescriptor.IsConst == 1 ? CV_PTR_IS_CONST : 0)));
+            }
+
+            return _nextTypeIndex++;
+        }
+
+        public uint GetArrayTypeIndex(
+            ClassTypeDescriptor classDescriptor,
+            ArrayTypeDescriptor arrayDescriptor)
+        {
+            uint memberCount = 0;
+            uint offset = 0;
+
+            using (LeafRecordWriter arrayRecord = StartLeafRecord(LF_ARRAY))
+            {
+                arrayRecord.Write(arrayDescriptor.ElementType);
+                arrayRecord.Write(T_INT4);
+                arrayRecord.Write(arrayDescriptor.Size);
+                arrayRecord.Write("");
+            }
+
+            uint arrayRecordTypeIndex = _nextTypeIndex++;
+
+            using (LeafRecordWriter fieldListRecord = StartLeafRecord(LF_FIELDLIST))
+            {
+                if (classDescriptor.BaseClassId != 0)
+                {
+                    fieldListRecord.StartListEntry(LF_BCLASS);
+                    fieldListRecord.Write((ushort)0); // TODO: Attributes
+                    fieldListRecord.Write(classDescriptor.BaseClassId);
+                    fieldListRecord.WriteEncodedInteger(0); // Offset
+                    fieldListRecord.EndListEntry();
+                    memberCount++;
+                    offset += (uint)_targetPointerSize;
+                }
+
+                fieldListRecord.StartListEntry(LF_MEMBER);
+                fieldListRecord.Write((ushort)0); // TODO: Attributes
+                fieldListRecord.Write(T_INT4);
+                fieldListRecord.WriteEncodedInteger(offset);
+                fieldListRecord.Write("count");
+                fieldListRecord.EndListEntry();
+                memberCount++;
+                offset += (uint)_targetPointerSize;
+
+                if (arrayDescriptor.IsMultiDimensional == 1)
+                {
+                    for (uint i = 0; i < arrayDescriptor.Rank; ++i)
+                    {
+                        fieldListRecord.StartListEntry(LF_MEMBER);
+                        fieldListRecord.Write((ushort)0); // TODO: Attributes
```

Comment:
Do these TODOs exist in the existing objwriter too?

Reply:
Good catch. They don't, the old ObjWriter uses `MemberAccess::Public` in most cases which likely translates to the value `3`. I'll go over the TODOs and make sure that correct attributes are used.

Reply:
I updated it. The values are still pretty bogus (ie. everything is "public") but they should match the output of the LLVM ObjWriter.